File: OCH_Tao/ABC/B/371.py
N,M = map(int,input().split())
L = [True]*N
for i in range(M):
  A,B = input().split()
  A = int(A)-1
  if B=="M" and L[A]:
    print("Yes")
    L[A]=False
  else:
    print("No")

File: OCH_Tao/ABC/B/371.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N,] = input.shift().split(" ").map(x => parseInt(x, 10));
let L = Array(N).fill(true);
for (const i of input) {
  const [A, B] = i.split(" ");
  if (B === "M" && L[parseInt(A, 10) - 1]) {
    console.log("Yes");
    L[parseInt(A, 10) - 1] = false;
  } else {
    console.log("No");
  }
}